src/compiler/types.rs

```diff
@@ -713,6 +730,85 @@ pub struct ExpressionStatement {
     pub expression: Rc</*Expression*/ Node>,
 }
 
+#[derive(Debug)]
+#[ast_type(interfaces = "NamedDeclarationInterface")]
+pub enum TypeElement {
+    PropertySignature(PropertySignature),
+}
+
+#[derive(Debug)]
+#[ast_type(ancestors = "TypeElement", interfaces = "NamedDeclarationInterface")]
+pub struct PropertySignature {
+    _named_declaration: BaseNamedDeclaration, /*name: PropertyName*/
+    pub type_: Option<Rc<Node /*TypeNode*/>>,
+}
+
+impl PropertySignature {
+    pub fn new(base_named_declaration: BaseNamedDeclaration, type_: Option<Rc<Node>>) -> Self {
+        Self {
+            _named_declaration: base_named_declaration,
+            type_,
+        }
+    }
+}
+
+impl HasTypeInterface for PropertySignature {
+    fn type_(&self) -> Option<Rc<Node>> {
+        self.type_.clone()
+    }
+
+    fn set_type(&mut self, type_: Rc<Node>) {
+        self.type_ = Some(type_);
+    }
+}
+
+#[derive(Debug)]
+#[ast_type(impl_from = false, interfaces = "NamedDeclarationInterface")]
```

Comment:
Is it required that the things inside the parentheses be key-value pairs? If `impl_from` is only used when you set it to false, I think having an `omit_impl_from` flag might be a little more idiomatic (from my limited use of fancy macros).

Reply:
Ya that makes sense, I'm not sure what the combination of `syn` + `darling` supports out-of-the-box